auto_nag/scripts/reminder.py

```diff
@@ -0,0 +1,146 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import re
+
+from dateutil import parser
+from libmozdata.bugzilla import BugzillaUser
+from libmozdata import utils as lmdutils
+
+from auto_nag import utils
+from auto_nag.bzcleaner import BzCleaner
+
+REMINDER_TYPES = [
+    "test",
+    "pref",
+]
+DATE_MATCH = re.compile(
+    r"\[reminder\-(%s) ([0-9]{4}-[0-9]{1,2}-[0-9]{1,2})\]" % "|".join(REMINDER_TYPES)
+)
+
+
+class Reminder(BzCleaner):
+    def __init__(self):
+        super(Reminder, self).__init__()
+        self.extra_ni = {}
+        self.autofix_whiteboard = {}
+
+    def description(self):
+        return "Bugs with whiteboard reminders"
+
+    def get_bz_params(self, date):
+        self.today = lmdutils.get_date_ymd(date)
+
+        params = {
+            "include_fields": ["assigned_to", "whiteboard", "triage_owner", "history"],
+            "f1": "status_whiteboard",
+            "o1": "substring",
+            "v1": "[reminder-",
+        }
+
+        return params
+
+    def handle_bug(self, bug, data):
+        bugid = str(bug["id"])
+
+        new_whiteboard = whiteboard = bug["whiteboard"]
+        matches = DATE_MATCH.findall(whiteboard)
+
+        # We support having multiple reminders that expire on a single day.
+        # (I'm not sure why, but I guess we should.)
+        reminders = []
+
+        for m in matches:
+            tag, date = m
+
+            target = parser.parse(date)
+
+            if target.date() < self.today.date():
+                replace_string = f"[reminder-{tag} {date}]"
+                new_whiteboard = new_whiteboard.replace(replace_string, "")
+
+                reminders.append({"full_tag": replace_string})
+
+        if new_whiteboard == whiteboard:
+            return
+
+        target_entries = []
+        for entry in bug["history"]:
+            for field in entry["changes"]:
+                if field["field_name"] == "whiteboard":
+                    # Check to see if any of the replace strings appeared in this change.
+                    for r in reminders:
+                        if (
+                            r["full_tag"] in field["added"]
+                            and r["full_tag"] not in field["removed"]
+                        ):
+                            entry["full_tag"] = r["full_tag"]
+                            target_entries.append(entry)
+
+        user_emails_to_names = self._get_user_emails_to_names(target_entries)
+
+        for r in reminders:
+            for entry in target_entries:
+                if r["full_tag"] == entry["full_tag"]:
+                    r["who"] = user_emails_to_names[entry["who"]]
+                    r["when"] = self._datestr(entry["when"])
```

Comment:
```suggestion
                    r["when"] = utils.get_human_lag(entry["when"])
```
Currently, the comment looks as follows:
>on Saturday, June 18 09:54 PM UTC, Suhaib Mujahid placed a reminder on the bug using the whiteboard tag `[reminder-test 2022-06-22]`

Using `get_human_lag()` it will looks as follows:

> 4 days ago, Suhaib Mujahid placed a reminder on the bug using the whiteboard tag `[reminder-test 2022-06-22]`


Note:  `get_human_lag()` does not add "ago", it need to be added in the template.